src/Mod/Draft/draftmake/make_polararray.py

```diff
@@ -131,4 +135,14 @@ def make_polar_array(base_object,
     new_obj = make_array.make_array(base_object,
                                     arg1=center, arg2=angle, arg3=number,
                                     use_link=use_link)
+    if axis_reference:
+        if not (isinstance(axis_reference, tuple) and isinstance(axis_reference[1], list) and len(axis_reference[1]) == 1):
+            _err(_tr("Wrong input. Must be ('DocumentObject', ['ObjectString'])"))
```

Comment:
This type of list `(object, ['Subelement'])` is called a `LinkSubList` in FreeCAD. It is common when you want to extract an internal element, `Vertex`, `Edge`, or `Face` from a `Shape`. However, it is not very user friendly to pass this to the `make_polar_array` because it is hard to remember the syntax. When you only have one subelement, it doesn't need to be inside a list, so it can be a simple two-element tuple or list, `(object, 'Subelement')` or `[object, 'Subelement']`.

I've been looking at other functions in Draft, and I think it's better if you pass two variables, an `axis_object`, and an `axis_edge`, then you can manually build the `LinkSubList` (`axis_reference`) from those two variables.

Look at `make_path_array` to see how I do it. This function accepts a `path_object` and `subelements`, and you build the underlying `PathSubelements` from those two.

So, it would be something like this
```py
def make_polar_array(..., axis_object=cube, axis_edge="Edge1", ...)
    ...
    axis_reference = []
    if axis_object and axis_edge:
        axis_reference = [axis_object, axis_edge]
    ...
    new_obj.AxisReference = axis_reference
```

If you don't specify the edge, then the function can choose a default edge, say the first one in the `Shape`.
```py
def make_polar_array(..., axis_object=cube, axis_edge=None, ...)
    ...
    axis_reference = []
    if axis_object and not axis_edge:
        if hasattr(axis_object, "Shape") and hasattr(axis_object.Shape, "Edges"):
            axis_reference = [axis_object, axis_object.Shape.Edges[0]]
        else:
            _wrn(_tr("Wrong input: axis_object cannot be used for Axis Reference, it lacks a Shape with Edges"))
    new_obj.AxisReference = axis_reference
```
This allows you to pass any object as `axis_object`, so it is more general. You can also do type checking on `axis_object` and `axis_edge` (string) this way, and the user doesn't have to think too much about the `LinkSubList` being in the correct format, he or she only needs to pass two things, `axis_object` and `axis_edge`.

If you see in `make_path_array`, `path_object` can be specified as an actual DocumentObject, or as a string. If it's a string, the function will assume it is a `Label`, and will search (`utils.find_object`) the corresponding object that has that label. This makes the function easier to use from the Python terminal, and less verbose.
```py
make_polar_array(..., axis_object=App.ActiveDocument.DatumLine, ...)  # uses default edge
make_polar_array(..., axis_object="DatumLine", ...)  # searches by Label and uses default edge
make_polar_array(..., axis_object=App.ActiveDocument.Box, axis_edge="Edge1", ...)
make_polar_array(..., axis_object="My funny box", axis_edge="Edge3", ...)  # searches by label and uses the specified edge
```

Reply:
It seems you updated your branch while I was writing this. Nevertheless, I hope the idea is more or less clear.

Just a further improvement, in my idea above, instead `axis_edge` being a string like `Edge1`, `Edge3`, etc., it could also be a simple integer. So, you'd add some logic so that `axis_edge=1` corresponds to `Edge1`, and `axis_edge=3` corresponds to `Edge3`, etc. However, do notice that the `LinkSubList` requires a string, so it would have to be converted.
```py
    axis_reference = []
    if axis_object and axis_edge and isinstance(axis_edge, int):
        axis_reference = [axis_object, "Edge" + str(axis_edge)]
    ...
    new_obj.AxisReference = axis_reference
```

I didn't do it like this in `make_path_array` in order to retain backwards compatibility, but this can still be changed for `make_polar_array` which has never had this option before.
```py
make_polar_array(..., axis_object="My funny box", axis_edge=3, ...)  # searches by label and uses the specified edge
```